codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -571,6 +570,22 @@ private void verifyStreamMayHaveExisted(int streamId) throws Http2Exception {
                 throw connectionError(PROTOCOL_ERROR, "Stream %d does not exist", streamId);
             }
         }
+
+        private void validateHeaders(int streamId, Http2Headers headers, Http2Stream stream, boolean endOfStream,
+                                     boolean isInformational) throws Http2Exception {
+            if (connection.isServer() &&
```

Comment:
Do we worry about the client side validation somewhere else?

Reply:
Baby steps, h2spec only validate server specification for now.